01_fib/main.go

```diff
@@ -0,0 +1,25 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fib(5)
+
+}
+
+func fib(n int) int {
+	var n1, n2, n3 = 0, 1, 0
+	for i := 0; i < n; i++ {
+		n1 = n2
+		n2 = n3
```

Comment:
You can make this a one-liner: `n1, n2 = n2, n3`.

Reply:
changed the logic to one-liner which was quite efficient